File: ios/GalleryListViewManager.m
//
//  GalleryListView.m
//  react-native-recyclerview-android
//
//  Created by wweng on 20/5/2023.
//

#import <Foundation/Foundation.h>
#import "React/RCTViewManager.h"
@interface RCT_EXTERN_MODULE(GalleryListViewManager, RCTViewManager)
RCT_EXPORT_VIEW_PROPERTY(dataSourceString, NSString)
RCT_EXPORT_VIEW_PROPERTY(sectionHeaderStyle, NSString)
RCT_EXPORT_VIEW_PROPERTY(recyclerPropString, NSString)
RCT_EXPORT_VIEW_PROPERTY(onLongPressed, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onItemPressed, RCTDirectEventBlock)
RCT_EXTERN_METHOD(toggleSelectionMode
                  : (nonnull NSNumber *)reactTag selectionMode
                  : (nonnull BOOL *)selectionMode)
@end

@interface RCT_EXTERN_MODULE(ZoomableImageViewManager, RCTViewManager)
@end

File: ios/GalleryListViewManager.swift
//
//  GalleryListView.swift
//  react-native-recyclerview-android
//
//  Created by wweng on 20/5/2023.
//
import Foundation
import React
@objc(GalleryListViewManager)
class GalleryListViewManager:RCTViewManager  {
    
    var galleryListView:GalleryListView!=nil;
    override func view() -> UIView! {
        
        let aRect = CGRect(x:0, y:0, width: 100, height: 100);
        galleryListView = GalleryListView(frame: aRect);
        return galleryListView;
    }
    
    override static func requiresMainQueueSetup() -> Bool {
        return true
    }
    
    @objc public func toggleSelectionMode(_ rectTag:NSNumber,selectionMode:Bool)-> Void {
        galleryListView.toggleSelectionMode(selectionMode);
    }
}
